search/views.py

```diff
@@ -0,0 +1,70 @@
+""" handle requests for courseware search http requests """
+import logging
+import json
+import datetime
+
+from django.http import HttpResponse
+from django.utils.translation import ugettext as _
+from django.views.decorators.http import require_POST
+
+from .api import perform_search
+
+# log appears to be standard name used for logger
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class DateTimeEncoder(json.JSONEncoder):
+
+    """ encode datetimes into json appropriately """
+
+    def default(self, obj):  # pylint: disable=method-hidden
+        """ override default encoding """
+        if isinstance(obj, datetime.datetime):
+            encoded_object = obj.isoformat()
+        else:
+            encoded_object = super(DateTimeEncoder, self).default(self, obj)
+        return encoded_object
+
+
+@require_POST
+def do_search(request, course_id=None):
+    """
+    Search view for http requests
+    """
```

Comment:
Can you list out all known POST parameters that can be passed in (and if they are required or optional) here?

Also, just so I understand the topology here. Are these HTTP views just for integration testing, or will the LMS (server or client code) be calling into these HTTP entry points? Or will LMS integrate to this subsystem as an "in-proc" library, or at least initially? Sorry, I guess I can look at the LMS-side code PR, but I figured I'd ask while I'm here.

In either case you might want to add a comment which use cases this HTTP endpoint is intended to support.